shiny/ui/_input_select.py

```diff
@@ -282,6 +340,17 @@ def _normalize_choices(x: SelectChoicesArg) -> _SelectChoices:
         return x
 
 
+def _contains_html(x: _SelectChoices) -> bool:
+    for v in x.values():
+        if isinstance(v, Mapping):
+            if _contains_html(cast(_OptGrpChoices, v)):
+                return True
```

Comment:
While the typings wouldn't recurse forever, the recursive nature of the function gives me a little pause.

Could we instead perform it directly as we know the recursive depth should only ever be 1 and we'd only check for "not strings"?

```suggestion
            # Check the `_Choices` values of `_OptGrpChoices`
            for vv in v.values():
                if not is instance(vv, str):
                    return True
```